modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1440,32 +1440,34 @@ def searchsorted(df):
 
     # Dt map partitions operations
 
-    dt_date = Map.register(_dt_prop_map("date"))
-    dt_time = Map.register(_dt_prop_map("time"))
-    dt_timetz = Map.register(_dt_prop_map("timetz"))
-    dt_year = Map.register(_dt_prop_map("year"))
-    dt_month = Map.register(_dt_prop_map("month"))
-    dt_day = Map.register(_dt_prop_map("day"))
-    dt_hour = Map.register(_dt_prop_map("hour"))
-    dt_minute = Map.register(_dt_prop_map("minute"))
-    dt_second = Map.register(_dt_prop_map("second"))
-    dt_microsecond = Map.register(_dt_prop_map("microsecond"))
-    dt_nanosecond = Map.register(_dt_prop_map("nanosecond"))
-    dt_week = Map.register(_dt_prop_map("week"))
-    dt_weekofyear = Map.register(_dt_prop_map("weekofyear"))
-    dt_dayofweek = Map.register(_dt_prop_map("dayofweek"))
-    dt_weekday = Map.register(_dt_prop_map("weekday"))
-    dt_dayofyear = Map.register(_dt_prop_map("dayofyear"))
-    dt_quarter = Map.register(_dt_prop_map("quarter"))
-    dt_is_month_start = Map.register(_dt_prop_map("is_month_start"))
-    dt_is_month_end = Map.register(_dt_prop_map("is_month_end"))
-    dt_is_quarter_start = Map.register(_dt_prop_map("is_quarter_start"))
-    dt_is_quarter_end = Map.register(_dt_prop_map("is_quarter_end"))
-    dt_is_year_start = Map.register(_dt_prop_map("is_year_start"))
-    dt_is_year_end = Map.register(_dt_prop_map("is_year_end"))
-    dt_is_leap_year = Map.register(_dt_prop_map("is_leap_year"))
-    dt_daysinmonth = Map.register(_dt_prop_map("daysinmonth"))
-    dt_days_in_month = Map.register(_dt_prop_map("days_in_month"))
+    dt_date = Map.register(_dt_prop_map("date"), dtypes=np.object_)
+    dt_time = Map.register(_dt_prop_map("time"), dtypes=np.object_)
+    dt_timetz = Map.register(_dt_prop_map("timetz"), dtypes=np.object_)
+    dt_year = Map.register(_dt_prop_map("year"), dtypes=np.int64)
+    dt_month = Map.register(_dt_prop_map("month"), dtypes=np.int64)
+    dt_day = Map.register(_dt_prop_map("day"), dtypes=np.int64)
+    dt_hour = Map.register(_dt_prop_map("hour"), dtypes=np.int64)
+    dt_minute = Map.register(_dt_prop_map("minute"), dtypes=np.int64)
+    dt_second = Map.register(_dt_prop_map("second"), dtypes=np.int64)
```

Comment:
could you please check this is correct? I mean, wasting int64 to store numbers from 0 to 59 feels a little too much...

Reply:
@vnlitvinov agree that it does seem a bit wasteful.

Reply:
this could probably be improved upstream.  the lower-level calls return int32, get wrapped somewhere in int64

Reply:
I checked using the following example, everything is so. Do you need to do something else or we can merge?
```python
pd.Series(pd.to_timedelta(np.arange(5), unit='d')).dt
```